com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaScanner.java

```diff
@@ -352,15 +358,18 @@ public void enterFunctioncall(LuaParser.FunctioncallContext ctx) {
             }
         }
         else if (fnDesc.is("property", "go")) {
+            ParserRuleContext paramsContext;
             LuaParser.ArgsContext argsCtx = ctx.nameAndArgs().args();
+            paramsContext = argsCtx;
             String firstArg = getFirstStringArg(argsCtx);
             List<Token> tokens = getTokens(ctx, Token.DEFAULT_CHANNEL);
             Property property = new Property(tokens.get(0).getLine() - 1);
             property.name = firstArg;
             if (firstArg == null) {
                 property.status = Status.INVALID_ARGS;
             } else {
-                if (parsePropertyValue(argsCtx, property)) {
+                paramsContext = parsePropertyValue(argsCtx, property);
```

Comment:
To be able to keep hash tokens, we need better return value than just a bool, taking into account what this function does, it's fine to return `ParserRuleContext` and use it for both the error's checking and getting tokens we wanna keep